staging/src/k8s.io/legacy-cloud-providers/azure/azure_client.go

```diff
@@ -308,8 +320,11 @@ func newAzInterfacesClient(config *azClientConfig) *azInterfacesClient {
 }
 
 func (az *azInterfacesClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, networkInterfaceName string, parameters network.Interface) (resp *http.Response, err error) {
+	mc := newMetricContext("interfaces", "create_or_update", resourceGroupName, az.client.SubscriptionID, "")
```

Comment:
since the prefix and request in `newMetricContext` are mostly repeated, we should define it these as constants and use it.

Reply:
Let us have this change as part of this PR please. Let us have a private var, create on first call `NewMetricContext()`

Reply:
As discussed, the metrics context is fresh for every operation.